db/views/reporting_overdue_patients_v01.sql

```diff
@@ -0,0 +1,194 @@
+WITH patients_with_appointments AS (
+    SELECT
+        DISTINCT ON (rps.patient_id, rps.month_date)
+        rps.month_date,
+        rps.patient_id,
+        rps.hypertension AS hypertension,
+        rps.diabetes AS diabetes,
+        rps.htn_care_state,
+        rps.month,
+        rps.quarter,
+        rps.year,
+        rps.month_string,
+        rps.quarter_string,
+        rps.assigned_facility_id,
+        rps.assigned_facility_slug,
+        rps.assigned_facility_region_id,
+        rps.assigned_block_slug,
+        rps.assigned_block_region_id,
+        rps.assigned_district_slug,
+        rps.assigned_district_region_id,
+        rps.assigned_state_slug,
+        rps.assigned_state_region_id,
+        rps.assigned_organization_slug,
+        rps.assigned_organization_region_id,
+        appointments.id AS previous_appointment_id,
+        appointments.device_created_at AS previous_appointment_date,
+        appointments.scheduled_date AS previous_appointment_schedule_date
+    FROM reporting_patient_states rps
+    LEFT JOIN appointments ON appointments.patient_id = rps.patient_id
+        AND appointments.device_created_at < rps.month_date
+    WHERE rps.status <> 'dead'
+	AND rps.month_date > NOW() - INTERVAL '24 months'
```

Comment:
If we are considering patients with appointments in the last 2 years, won't the specs set relative to june 2021 start failing soon?

Reply:
> If we are considering patients with appointments in the last 2 years, won't the specs set relative to June 2021 start failing soon?
Yeah, that's true. Thanks for pointing it out 👍

Reply:
> You can move the NOW() computation to a CTE to not compute this for every row (it'll save like 10s in BD).

@prabhanshuguptagit Good suggestion. Lemme test it out

Reply:
Thanks @qptr! The tests would have started to fail in 2 months.